autometa/common/external/diamond.py

```diff
@@ -29,146 +29,301 @@
         return self
 
     def get_top_hit(self):
-        top_bitscore = float('-Inf')
+        """
+        Returns the subject sequene ID with the highest bitscore amongst all the subject sequence that hit a query
+        """
+        top_bitscore = float("-Inf")
         top_hit = None
-        for sseqid,attrs in self.sseqids.items():
-            if attrs.get('bitscore', float('-Inf')) >= top_bitscore:
-                top_bitscore = attrs.get('bitscore')
+        for sseqid, attrs in self.sseqids.items():
+            if attrs.get("bitscore", float("-Inf")) >= top_bitscore:
+                top_bitscore = attrs.get("bitscore")
                 top_hit = sseqid
         return top_hit
 
-def makedatabase(fasta, database, nproc=1):
-    cmd = f'diamond makedb --in {fasta} --db {database} -p {nproc}'
-    logger.debug(f'{cmd}')
-    with open(os.devnull, 'w') as stdout, open(os.devnull, 'w') as stderr:
-        retcode = subprocess.call(cmd, stdout=stdout, stderr=stderr, shell=True)
-    if retcode:
-        raise OSError(f'DiamondFailed:\nArgs:{proc.args}\nReturnCode:{proc.returncode}')
+
+def makedatabase(fasta, database, nproc=mp.cpu_count()):
+    """
+    Creates a database against which the query sequence would be blasted
+
+    Parameters
+    ----------
+    fasta : str
+        Path to fasta file whose database needs to be made
+        e.g. '<path/to/fasta/file>'
+    database : str
+        Path to the output  nr database file
+        e.g. '<path/to/database/file>'
+    nproc : int, optional
+        Number of processors to be used. By default uses all the processors of the system 
+
+    Returns
+    -------
+    str
+        Path to Diamond database
+
+    Raises
+    ------
+    CalledProcessError
+        Failed to create Diamond database
+    """
+    cmd = f"diamond makedb --in {fasta} --db {database} -p {nproc}"
+    logger.debug(f"{cmd}")
+    proc = subprocess.run(
+        cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, text=True
+    )
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise err
     return database
 
-def blast(fasta, database, outfpath, blast_type='blastp', evalue=float('1e-5'),
-    maxtargetseqs=200, cpus=0, tmpdir=os.curdir, force=False, verbose=False):
-    """Performs diamond blastp search using fasta against diamond formatted database
+
+def blast(
+    fasta,
+    database,
+    outfpath,
+    blast_type="blastp",
+    evalue=float("1e-5"),
+    maxtargetseqs=200,
+    nproc=mp.cpu_count(),
+    tmpdir=os.curdir,
+    force=False,
+    verbose=False,
+):
+    """
+    Performs diamond blastp search using query sequence against diamond formatted database
 
     Parameters
     ----------
     fasta : str
-        </path/to/fasta/file>. May be amino acid or nucleotide sequences
+        Path to fasta file having the query sequence. May be amino acid or nucleotide sequences
```

Comment:
The _fasta_ may contain multiple sequences. These would all be query sequences. The description suggests the fasta file should only contain one query sequence, which is false.